[ES] eBPF translated to Spanish

SullyMA opened the pull request:
### Describe your changes


### Related issue number or link (ex: `resolves #issue-number`)


### Checklist before opening this PR (put `x` in the checkboxes)
- [✔️] This PR does not contain plagiarism
  - don’t copy other people’s work unless you are quoting and contributing it to them.
- [✔️] I have signed off on all commits 
  - [signing off](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) (ex: `git commit -s`) is to affirm that commits comply [DCO](https://wiki.linuxfoundation.org/dco). If you are working locally, you could add an alias to your `gitconfig` by running `git config --global alias.ci "commit -s"`.

ramrodo reviewed (changes requested):
Hi @SullyMA, thanks for submitting this PR.

All the commits should be "signed-off" to have the tests in green. It seems there are 3 commits on this PR but just the last one has been signed. So, you can do a git ammend|reset the previous commits to fix this.

Please reach out to us if you need help with this :)

electrocucaracha commented on content/es/ebpf.md:
```diff
@@ -0,0 +1,41 @@
+Sin embargo, este enfoque también introduce riesgos de seguridad, lo que hace que eBPF sea una alternativa atractiva.
+
+## Problema que soluciona
+Típicamente, las aplicaciones se ejecutan en el espacio de usuario y, si la aplicación requiere algunos privilegios del kernel (por ejemplo, para acceder a algún hardware), lo solicita al kernel a través de una llamada al sistema conocida como "llamada al sistema".  
+
+En la mayoría de los casos, este enfoque funciona bien. Sin embargo, hay casos en los que los desarrolladores requieren más flexibilidad para acceder al sistema a nivel bajo.
+
+La observabilidad, la seguridad y las características de redes son buenos ejemplos.
+
+Para lograrlo, podemos utilizar extensiones del kernel de Linux, ampliando la base del kernel sin modificar el código principal del kernel. 
+
+Si bien hay beneficios en el uso de extensiones del kernel de Linux, también introduce riesgos de seguridad. 
+
+Como operan dentro del espacio del kernel, las extensiones del kernel de Linux pueden hacer que el kernel falle y, cuando el kernel falla, también lo hace toda la máquina.
+
+Además, las extensiones del kernel tienen privilegios elevados y acceso directo a los recursos del sistema. Y si no están adecuadamente aseguradas, los atacantes pueden explotarlas.
```
```suggestion
Típicamente, las aplicaciones se ejecutan en el espacio de usuario y, si la aplicación requiere algunos privilegios del kernel (por ejemplo, para acceder a algún hardware),
lo solicita al kernel a través de una llamada al sistema conocida como "llamada al sistema".  
En la mayoría de los casos, este enfoque funciona bien. Sin embargo, hay casos en los que los desarrolladores requieren más flexibilidad para acceder al sistema a nivel bajo.
La observabilidad, la seguridad y las redes son buenos ejemplos.
Para lograrlo, podemos utilizar extensiones del kernel de Linux, ampliando la base del kernel sin modificar el código principal del kernel. 
Si bien hay beneficios en el uso de extensiones del kernel de Linux, también introduce riesgos de seguridad. 
Como operan dentro del espacio del kernel, las extensiones del kernel de Linux pueden hacer que el kernel falle y, cuando el kernel falla, también lo hace toda la máquina.
Además, las extensiones del kernel tienen privilegios elevados y acceso directo a los recursos del sistema. Y si no están adecuadamente aseguradas, los atacantes pueden explotarlas.
```

SullyMA replied:
The translation looks identical - are we just removing the white spaces here?

electrocucaracha commented on content/es/ebpf.md:
```diff
@@ -0,0 +1,41 @@
+Se ejecuta en un entorno aislado dentro del kernel, proporcionando aislamiento y mitigando riesgos. 
+Si se explota una vulnerabilidad o defecto en un programa eBPF, su impacto generalmente se limita al entorno aislado.
+Además, antes de que un programa eBPF pueda comenzar a ejecutarse en el kernel, debe pasar por algunas verificaciones. 
+El componente verificador verifica el programa eBPF en busca de posibles violaciones de seguridad, como el acceso a memoria fuera de límites, ciclos infinitos y funciones del kernel no autorizadas.
```
```suggestion
El componente verificador verifica el programa eBPF en busca de posibles violaciones de seguridad,
como el acceso a memoria fuera de límites, ciclos infinitos y funciones del kernel no autorizadas.
```

jossemarGT commented on content/es/ebpf.md:
```diff
@@ -0,0 +1,41 @@
+etiquetas: ["arquitectura", "redes", "seguridad"]
+---
+
+eBPF, o Filtro de Paquetes Berkeley Extendido, es una tecnología que permite que programas o scripts pequeños y aislados se ejecuten en el espacio del kernel de un sistema Linux sin necesidad de cambiar el código del kernel ni cargar extensiones del kernel de Linux.
```
```suggestion
eBPF, o "extended Berkeley Packet Filter", es una tecnología que permite a pequeños programas o scripts aislados ejecutarse en el espacio del kernel de un sistema Linux sin necesidad de cambiar el código fuente ni cargar módulos al kernel de Linux.
```

Otras traducciones han decidido utilizar "extended Berkeley Packet Filter" como si fuera un nombre propio y no traducirlo.

jossemarGT commented on content/es/ebpf.md:
```diff
@@ -0,0 +1,41 @@
+
+eBPF, o Filtro de Paquetes Berkeley Extendido, es una tecnología que permite que programas o scripts pequeños y aislados se ejecuten en el espacio del kernel de un sistema Linux sin necesidad de cambiar el código del kernel ni cargar extensiones del kernel de Linux.
+
+Un sistema Linux tiene dos espacios: el kernel y el espacio de usuario. 
```
```suggestion
Un sistema Linux tiene dos espacios: el de kernel y el de usuario. 
```

SullyMA commented on content/es/ebpf.md:
```diff
@@ -0,0 +1,41 @@
+
+Las aplicaciones quedan en el espacio de usuario y, cuando necesitan permisos más elevados, envían una solicitud al kernel.
+
+Para aplicaciones que requieren más flexibilidad, como el acceso directo al hardware, el kernel puede ser extendido mediante lo que se conoce como el enfoque de "extensiones del kernel de Linux". Este enfoque amplía la funcionalidad predeterminada del kernel, permitiendo a las aplicaciones un acceso más profundo a los componentes básicos. 
+
```
The translation looks identical - are we just removing the white spaces here?

electrocucaracha replied:
That's right, this suggestion includes some line breaks for matching the lines of the English version and easily identify differences.

jossemarGT commented on content/es/ebpf.md:
```diff
@@ -0,0 +1,41 @@
+título: eBPF
+estado: Completado
+categoría: Tecnología
+etiquetas: ["arquitectura", "redes", "seguridad"]
```
It's tricky, I know but you can see the section between the `---` like something that both human and machine "reads". That's why everything on the side of the `:` has to stay the same, since the software will look for those specific keywords, while the right size can be translated (with some exceptions like `status`).

electrocucaracha replied:
My understanding is because there is a [page](https://glossary.cncf.io/es/tags/) where you can search concepts by category, so that's why those terms have to be also translated but keeping the keys as they are in English

jossemarGT reviewed (changes requested):
I consider we should use _módulo/módulos_ instead _extensión/extensiones_ for modules, since the latter term could blur the distinction between modules and eBPF scripts in Spanish.

ramrodo reviewed (approved):
Great! Thanks for this contribution!

ramrodo reviewed (changes requested):
Hi @SullyMA, it seems that the base branch is `main` but it should be `dev-es`. Could you change this, please?

SullyMA commented:
> Hi @SullyMA, it seems that the base branch is `main` but it should be `dev-es`. Could you change this, please?

SullyMA commented:
I accidentally closed the pull. I meant to add a comment instead. I switched from main to dev-es. Let me know if these changes are reflected on your end. I do think Victor also went ahead and updated it.

ramrodo reviewed (commented):
LGTM. We only need to add the English words to the whitelist to have the Spellcheck test in green.

ramrodo commented on content/es/.wordlist.txt:
```diff
@@ -1,4 +1,5 @@
-add
+Here's the updated list with the words you provided added in alphabetical order:
+
```
Remove this sentence.

```suggestion
Here's the updated list with the words you provided added in alphabetical order:

```

ramrodo commented on content/es/.wordlist.txt:
```diff
@@ -34,6 +36,7 @@ CaaS
 calendar
 Canary
 CAPEX
+call
```
The word "call" should be between the words "calendar" and "Canary"

ramrodo commented on content/es/.wordlist.txt:
```diff
@@ -97,6 +100,7 @@ draft
 Dropbox
 edit
 ej
+eBPF
```
The word "eBPF" should be between the words "Dropbox" and "edit"

ramrodo commented on content/es/.wordlist.txt:
```diff
@@ -114,6 +118,7 @@ FaaS
 faltantes
 Feedback
 Finance
+Filter
```
The word "Filter" should be between the words "Feedback" and "Finance"

ramrodo commented on content/es/.wordlist.txt:
```diff
@@ -216,6 +221,7 @@ Open
 PaaS
 PaC
 Paganini
+packet
```
The word "packet" should be between the words "PaC" and "Paganini"

ramrodo reviewed (changes requested):
The new words should be in alphabetical order compared with the exiting ones on the list.

You can sort the list automatically with the following command:

`LC_ALL="" sort -o content/es/.wordlist.txt content/es/.wordlist.txt`

ramrodo commented on .gitpod.yml:
Remove this file

ramrodo reviewed (approved):
Great! Thanks, Sully!
